packages/loader/container-loader/src/container.ts

```diff
@@ -482,6 +486,26 @@ export class Container extends EventEmitterWithErrorHandling<IContainerEvents> i
                 }
             });
         }
+
+        // We observed that most users of platform do not check Container.connected event on load, causing bugs.
+        // As such, we are raising events when new listener pops up.
+        // Note that we can raise both "disconnected" & "connect" events at the same time,
+        // if we are in connecting stage.
+        this.on("newListener", (event: string, listener: (...args: any[]) => void) => {
+            // Fire events on the end of JS turn, giving a chance for caller to be in consistent state.
+            Promise.resolve().then(() => {
+                const connected = this._connectionState === ConnectionState.Connected;
+                if (event === connectedEventName && connected) {
+                    listener(event, this.clientId);
+                } else if (event === disconnectedEventName && !connected) {
+                    listener(event);
+                } else if (event === connectEventName && this._connectionState !== ConnectionState.Disconnected) {
+                    listener(event);
+                }
+            }).catch((error) =>  {
+                this.logger.sendErrorEvent({ eventName: "RaiseConnectedEventError" }, error);
```

Comment:
can we include the container clientid here like we do in raiseConnectedEvent

Reply:
I'm not sure it's needed here. The purpose of this event is to bring attention to logic error in code. It has nothing to do with clientId, and clientId can be easily found in the steam of events for this session, so IMHO it does not belong here.